specs/order_spec.rb

```diff
@@ -76,35 +81,105 @@
       result.must_equal true
     end
   end
+
+  describe "#remove_product" do
+    it "decreases the number of products" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_count = products.count
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("banana")
+      expected_count = before_count - 1
+      order.products.count.must_equal expected_count
+    end
+
+    it "returns true if the product is removed" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("banana")
+      # order.products.include?("banana").must_equal false
+      order.products.wont_include("banana")
+    end
+
+    it "returns false if the product is not removed" do
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::Order.new(1337, products)
+
+      order.remove_product("cracker")
+      order.products.include?("banana").must_equal true
+    end
+  end
 end
 
 # TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      # arrange
+      # NO arrange needed here
+      # act
+      Grocery::Order.all
+      # assert
+      Grocery::Order.all.must_be_kind_of Array
     end
 
     it "Returns accurate information about the first order" do
-      # TODO: Your test code here!
+      # arrange
+      # NO arrange needed here
+      # act
+      Grocery::Order.all[0].id
+      # assert
+      Grocery::Order.all[0].id.must_equal "1"
     end
 
     it "Returns accurate information about the last order" do
-      # TODO: Your test code here!
+      # arrange
+      # NO arrange needed here
+      # act
+      Grocery::Order.all[-1].id
+      # assert
+      Grocery::Order.all[-1].id.must_equal CSV.read('support/orders.csv', 'r')[-1][0]
     end
   end
 
   describe "Order.find" do
     it "Can find the first order from the CSV" do
-      # TODO: Your test code here!
+      #arrange
+      # N/A
+      # act
+      Grocery::Order.find("first")
```

Comment:
It would be better to find the first Order of the file by the ID field of that order. 

Something like: `Grocery::Order.find(1)`